web/src/api/auth/login-callback.ts

```diff
@@ -54,13 +56,21 @@ export const auth0Login = withApiAuthRequired(
 
     if (isEmailUser(auth0User)) {
       try {
-        const userData = await FetchUserByAuth0IdSdk(client).FetchUserByAuth0Id(
-          {
-            auth0Id: auth0User.sub,
-          }
-        );
+        const userData = await FetchUserByAuth0IdSdk(client).FetchEmailUser({
+          auth0Id: auth0User.sub,
+          email: auth0User.email,
+        });
 
-        user = userData?.user[0];
+        if (userData.userByAuth0Id.length > 0) {
+          user = userData.userByAuth0Id[0];
+        }
+
+        if (
+          userData.userByAuth0Id.length === 0 &&
+          userData.userByEmail.length > 0
+        ) {
+          user = userData.userByEmail[0];
+        }
```

Comment:
the logic here is not following [the specs](https://www.notion.so/worldcoin/WDP17-Auth0-Dev-Portal-Login-d1cff5fc500746578deac731da5d52c1?pvs=4#813cdf8ad0284c5f8d7a54519102c996) exactly, we should check for auth0Id first (takes precedence), then nullifier hash, then email. with the current logic we're checking nullifier hash first for users from Sign in with Worldcoin

Reply:
if we want to prioritize requests like you suggest, then we'll need to make two requests:
1. Fetch user by auth0
2. in case there is no data returned we making request by email or by nullifier. 
3. Then mutation to add auth0 
4. mutation to sync name + email.

Reply:
we could do a single GraphQL query to fetch user by Auth0Id, nullifier hash & email (separately as you have it). we then set the `user` variable based on the first result from the GraphQL query. once we have the user object we check whether we need to update name, email and/or Auth0Id. so it's minimum 1 request, max 2. thoughts?